### PR TITLE
Add optional paramaters to functions

### LINKED ISSUE #8: API requests should have optional parameters enabled

### PR DESCRIPTION
**Pull request by nph4rd**
There are some functions that do not have the optional parameters enabled. 

This PR will add them. 

Closes: #8

### CODE REVIEW

File: src/client.rs [hunk 1]
```diff
@@ -55,6 +55,24 @@ pub enum ApiError {
...
 
+/// Generic optional parameters for methods
+pub struct OptionalParams<'a> {
+    pub marker: Option<&'a u32>,
+    pub sort: Option<&'a str>,
+    pub limit: Option<&'a u8>,
+}
+
+/// Optional parameters for an order.
+/// For more info see: <https://bitso.com/api_info?python#place-an-order>
```
Comment (mucinoab) [inline src/client.rs]: I think we could remove the "python" from the url. 
"https://bitso.com/api_info?place-an-order#place-an-order"

File: src/client.rs [hunk 2]
```diff
@@ -486,20 +648,32 @@ impl Bitso {
             }
             None => return Err(anyhow!(EMPTY_CREDENTIALS_MSG)),
         }
-        let result = self
-            .get(&url, &mut HashMap::new(), ApiType::Private)
-            .await?;
+        println!("{:?}", url);
```
Comment (mucinoab) [inline src/client.rs]: Debugging?

**Review by mucinoab — COMMENTED** [review]
LGTM but I think it would be more ergonomic to have the `OptionalOrderParams` and such wrapped in an `Option` (`optional_params: Option<OptionalParams<'_>>`)  because if you don't need any optional parameters you just have to pass `None` instead of having to construct the whole structure.

**Review by mucinoab — DISMISSED** [review]

File: src/client.rs (hunk 1 above)
Comment (nph4rd) [reply src/client.rs]: Ah! Thank you!

File: src/client.rs (hunk 2 above)
Comment (nph4rd) [reply src/client.rs]: Indeed :) Thank you!

**Comment by nph4rd** [comment]
> LGTM but I think it would be more ergonomic to have the `OptionalOrderParams` and such wrapped in an `Option` (`optional_params: Option<OptionalParams<'_>>`) because if you don't need any optional parameters you just have to pass `None` instead of having to construct the whole structure.

This is a really good point! I'll make this change before merging.

**Comment by nph4rd** [comment]
@mucinoab - I've made the changes you suggested. I also included a more explicit construction of the parameter `Map` when there are optional parameters included so as not to include any empty parameters in the body. Let me know what you think.

**Review by mucinoab — APPROVED** [review]

**Pull request merged**